client/client.go

```diff
@@ -84,7 +86,22 @@ func makeClient(cfg clientConfig) (Client, error) {
 		}
 	}
 
-	return newWatchAggregator(c, cfg.log), nil
+	c = newWatchAggregator(c, cfg.log)
+
+	if cfg.prometheus != nil {
+		if cfg.id == "" {
```

Comment:
i would expect the registry passed in would use
use [`WrapRegistererWith`](https://godoc.org/github.com/prometheus/client_golang/prometheus#WrapRegistererWith) rather than this needing to be in the interface